custom/icds_reports/utils/__init__.py

```diff
@@ -1774,3 +1777,81 @@ def create_group_by(aggregation_level):
 
 def column_value_as_per_agg_level(aggregation_level, agg_level_threshold, true_value, false_value):
     return true_value if aggregation_level > agg_level_threshold else false_value
+
+
+@attr.s
+class AggLevelInfo(object):
+    agg_level = attr.ib()
+    col_name = attr.ib()
+
+
+def _construct_repacement_map_from_awc_location(loc_level, replacement_location_ids):
+    levels = {
+        'state': AggLevelInfo(1, 'state'),
+        'district': AggLevelInfo(2, 'district'),
+        'block': AggLevelInfo(3, 'block'),
+        'supervisor': AggLevelInfo(4, 'supervisor'),
+        'awc': AggLevelInfo(5, 'doc')
+    }
+    level_info = levels[loc_level]
+    filters = {
+        f'{level_info.col_name}_id__in': replacement_location_ids,
+        'aggregation_level': level_info.agg_level
+    }
+    columns = [
+        f'{level_info.col_name}_id',
+        'state_name',
+        'district_name',
+        'block_name',
+        'supervisor_name',
+        'awc_name'
+    ]
+    replacements = AwcLocation.objects.filter(**filters).values(*columns)
+
+    def _full_hierarchy_name(loc):
```

Comment:
is it better to have inline functions declared on top?

Reply:
I declared it here because it was used once and this is right before that use. I could be convinced it should be higher up

Reply:
Seeing other instances on HQ i see they are declared in the beginning. Does that convince you?

Reply:
good enough for me

Reply:
moved